src/main/java/com/example/basejook/repository/StudentRepository.java

```diff
@@ -2,22 +2,48 @@
 
 import com.example.basejook.service.domain.Student;
 import lombok.RequiredArgsConstructor;
+import org.jooq.DSLContext;
 import org.springframework.stereotype.Repository;
+import org.jooq.your.base.tables.records.StudentRecord;
+import java.util.Random;
+
+import static org.jooq.your.base.Tables.STUDENT;
 
 @Repository
 @RequiredArgsConstructor
 
 public class StudentRepository {
+    private  final DSLContext dsl;
+
     public Student getById (Long id){
-        return null;
+        StudentRecord student = dsl
+                .selectFrom(STUDENT)
+                .where(STUDENT.ID.eq(id.intValue()))
+                .fetchOne();
+        return  Student.builder()
+                .id(Long.valueOf(student.getId()))
+                .name(student.getFirstname())
+                .surname(student.getSecondname())
+                .build();
+
     }
     public Student create (Student student){
-        return null;
+        Random random = new Random();
+        dsl.insertInto(STUDENT, STUDENT.ID, STUDENT.FIRSTNAME, STUDENT.SECONDNAME)
+                .values(random.nextInt(), student.getName(), student.getSurname())
+                .execute();
+        return student;
     }
     public Student update (Long id, Student student){
-        return null;
+        dsl.update(STUDENT)
+                .set(STUDENT.FIRSTNAME, student.getName())
+                .set(STUDENT.SECONDNAME, student.getSurname())
+                .where(STUDENT.ID.eq(id.intValue()))
```

Comment:
can you generate this ID in jook like Long?

Reply:
no because ID in table have a type Integer